metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1422,11 +1422,17 @@ void handleBrokerInControlledShutdown(int brokerId, long brokerEpoch, List<ApiMe
      * @param records       The record list to append to.
      */
     void handleBrokerUncleanShutdown(int brokerId, List<ApiMessageAndVersion> records) {
-        if (!featureControl.metadataVersion().isElrSupported()) return;
-        generateLeaderAndIsrUpdates("handleBrokerUncleanShutdown", NO_LEADER, NO_LEADER, brokerId, records,
-            brokersToIsrs.partitionsWithBrokerInIsr(brokerId));
-        generateLeaderAndIsrUpdates("handleBrokerUncleanShutdown", NO_LEADER, NO_LEADER, brokerId, records,
-            brokersToElrs.partitionsWithBrokerInElr(brokerId));
+        if (!featureControl.metadataVersion().isElrSupported()) {
```

Comment:
This is a nit, but it would be nice to remove the ! here and just put the ELR case first. Since we now have both cases handled.